File: Chess/DataTypes/CardDeck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataTypes
{
    public class CardDeck
    {
        public List<Card> Deck = new List<Card>();
        public Stack<Card> GameDeck = new Stack<Card>();

        public CardDeck()
        {
            for(int i=0; i<4; i++)
            {
                for(int j=0; j<13; j++)
                {
                    Deck.Add(new Card((Lear)i, (Rating)j));
                }
            }
        }

        public void MakeMixList()
        {
            Random rand = new Random();
            SortedList<int, Card> mixedList = new SortedList<int, Card>();
            foreach (Card item in Deck)
                mixedList.Add(rand.Next(), item.Clone());
            for (int i = 0; i < mixedList.Count; i++)
            {
                GameDeck.Push(mixedList.Values[i]);
            }
        }

        public Card TopCard()
        {
            return GameDeck.Pop();
        }

        public void Discard()
        {
            GameDeck.Pop();
        }

        public Card[] GetHand()
        {
            return new Card[] { TopCard(), TopCard() };
        }

        public Card[] Get5Cards()
        {
            return new Card[] { TopCard(), TopCard(), TopCard(), TopCard(), TopCard() };
        }
    }
}

File: Chess/NodeChessServer/DataTypes/CardDeck.js
const Card = require('./Card');

class CardDeck {
    constructor() {
        this.deck = [];
        for (let lear = 0; lear < 4; lear++) {
            for (let rating = 2; rating <= 14; rating++) {
                this.deck.push(new Card(lear, rating));
            }
        }
        this.MakeMixList();
    }

    MakeMixList() {
        this.gameDeck = [];
        for (const card of this.deck) {
            this.gameDeck.push(card);
        }
        this.gameDeck.sort(() => (5 - Math.random() * 10));
    }

    TopCard() {
        return this.gameDeck.pop();
    }

    Discard() {
        this.gameDeck.pop();
    }

    GetHand() {
        return [
            this.TopCard(),
            this.TopCard()
        ];
    }

    GetFloop() {
        return [
            this.TopCard(),
            this.TopCard(),
            this.TopCard()
        ];
    }
}

module.exports = CardDeck;